src/Client.php

```diff
@@ -179,104 +179,145 @@ public function delete(string $key)
      */
-    public function putIf(string $key, $value, $previousValue, bool $returnNewValueOnFail = false)
+    public function putIf(string $key, string $value, $compareValue, bool $returnNewValueOnFail = false)
     {
-        $request = new PutRequest();
-        $request->setKey($key);
-        $request->setValue($value);
-
-        $operation = new RequestOp();
-        $operation->setRequestPut($request);
+        $operation = $this->getPutOperation($key, $value);
+        $compare = $this->getCompareForIf($key, $compareValue);
+        $failOperation = $this->getFailOperation($key, $returnNewValueOnFail);
 
-        return $this->requestIf($key, $previousValue, $operation, $returnNewValueOnFail);
+        $response = $this->requestIf($key, [$operation], $failOperation, [$compare]);
+        return $this->getIfResponse($returnNewValueOnFail, $response);
     }
 
     /**
      * Delete if $key value matches $previous value otherwise $returnNewValueOnFail
      *
      * @param string $key
-     * @param $previousValue
+     * @param bool|string $compareValue The previous value to compare against
      * @param bool $returnNewValueOnFail
      * @return bool|string
      * @throws InvalidResponseStatusCodeException
+     * @throws \Exception
      */
-    public function deleteIf(string $key, $previousValue, bool $returnNewValueOnFail = false)
+    public function deleteIf(string $key, $compareValue, bool $returnNewValueOnFail = false)
     {
-        $request = new DeleteRangeRequest();
-        $request->setKey($key);
+        $operation = $this->getDeleteOperation($key);
+        $compare = $this->getCompareForIf($key, $compareValue);
+        $failOperation = $this->getFailOperation($key, $returnNewValueOnFail);
 
-        $operation = new RequestOp();
-        $operation->setRequestDeleteRange($request);
-
-        return $this->requestIf($key, $previousValue, $operation, $returnNewValueOnFail);
+        $response = $this->requestIf($key, [$operation], $failOperation, [$compare]);
+        return $this->getIfResponse($returnNewValueOnFail, $response);
     }
 
     /**
      * Execute $requestOperation if $key value matches $previous otherwise $returnNewValueOnFail
      *
      * @param string $key
-     * @param $previousValue
-     * @param RequestOp $requestOperation
+     * @param array $requestOperations operations to perform on success, array of RequestOp objects
+     * @param array|null $failureOperations operations to perform on failure, array of RequestOp objects
+     * @param array $compare array of Compare objects
      * @param bool $returnNewValueOnFail
-     * @return bool|string
+     * @return TxnResponse
      * @throws InvalidResponseStatusCodeException
      */
-    protected function requestIf(string $key, $previousValue, RequestOp $requestOperation, bool $returnNewValueOnFail = false)
+    public function requestIf(string $key, array $requestOperations, ?array $failureOperations, array $compare): TxnResponse
```

Comment:
```suggestion
    public function txnRequest(string $key, array $requestOperations, ?array $failureOperations, array $compare): TxnResponse
```

I think the name fits no longer and would suggest `txnRequest` instead.